static/app/views/settings/organizationIntegrations/configureIntegration.spec.tsx

```diff
@@ -0,0 +1,142 @@
+            },
+          },
+        ],
+      },
+    });
+
+    MockApiClient.addMockResponse({
+      url: `/organizations/${org.slug}/integrations/${integrationId}/`,
+      body: {
+        id: '1',
+        name: 'hello-world',
+        icon: null,
+        domainName: 'hello-world.app.opsgenie.com',
+        accountType: null,
+        scopes: null,
+        status: 'active',
+        provider: {
+          key: 'opsgenie',
+          slug: 'opsgenie',
+          name: 'Opsgenie (Integration)',
+          canAdd: true,
+          canDisable: false,
+          features: ['alert-rule', 'incident-management'],
+          aspects: {},
+        },
+        configOrganization: [
+          {
+            name: 'team_table',
+            type: 'table',
+            label: 'Opsgenie integrations',
+            help: "If integration keys need to be updated, deleted, or added manually please do so here. Your keys must be associated with a 'Sentry' Integration in Opsgenie.                 Alert rules will need to be individually updated for any key additions or deletions.",
+            addButtonText: '',
+            columnLabels: {
+              team: 'Label',
+              integration_key: 'Integration Key',
+            },
+            columnKeys: ['team', 'integration_key'],
+            confirmDeleteMessage:
+              'Any alert rules associated with this integration will stop working. The rules will still exist but will show a `removed` team.',
+          },
+        ],
+        configData: {
+          team_table: [
+            {
+              team: 'python [MIGRATED]',
+              id: '1-python',
+              integration_key: 'abcdef',
+            },
+          ],
+        },
+        externalId: 'hello-world',
+        organizationId: 1,
+        organizationIntegrationStatus: 'active',
+        gracePeriodEnd: null,
+      },
+    });
+
+    MockApiClient.addMockResponse({
+      url: `/organizations/${org.slug}/plugins/configs/`,
+      body: [
+        {
+          id: 'opsgenie',
+          name: 'Opsgenie',
+          slug: 'opsgenie',
+          projectList: [
+            {
+              projectId: 2,
+              projectSlug: 'python',
+              projectName: 'python',
+              enabled: true,
+              configured: true,
+              projectPlatform: 'python',
+            },
+          ],
+        },
+      ],
+    });
+    MockApiClient.addMockResponse({
+      url: `/organizations/${org.slug}/integrations/${integrationId}/migrate-opsgenie/`,
+      body: {},
+    });
+    render(
+      <ConfigureIntegration
+        {...TestStubs.routeComponentProps()}
+        params={{integrationId, providerKey: 'opsgenie'}}
+        organization={org}
+        location={TestStubs.location({query: {}})}
+      />
+    );
+    expect(screen.getByRole('button', {name: 'Migrate Plugin'})).toBeEnabled();
+
+    await userEvent.click(screen.getByRole('button', {name: 'Migrate Plugin'}));
```

Comment:
can you test that the `PUT` method was called?

Reply:
I've included a mock for the URL, unless I'm missing something.

Reply:
You can test that the mock was called by saving the call to a variable and checking its call history:


```javascript
const putIntegrationConfig = MockApiClient.addMockResponse({
    url: someUrl,
    method:'PUT',
    body: someResponseBody
})
// click some button
expect(putIntegrationConfig).toHaveBeenCalledWith(...)
// or just
expect(putIntegrationConfig).toHaveBeenCalled()
```